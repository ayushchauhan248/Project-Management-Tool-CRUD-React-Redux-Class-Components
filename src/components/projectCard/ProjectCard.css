/* #cardcvr{
    padding-left:3px;
    padding-top: 3px;
    padding-bottom: 3px;
    padding-right: 1px;
    /* z-index: -10; */
/* }  */

.card {
    padding: 10px;
    width: 700px;
    min-height: 100px;
    border-radius: 10px;
    transition: all 0.2s;
    background: rgb(80, 73, 119);
    margin: auto;
    
}

.card:hover {
    box-shadow: 0px 6px 10px rgba(0, 0, 0, 0.4);
}

.card__exit,
.card__icon {
    text-decoration: none;
    color: rgba(255, 255, 255, 0.9);
}

.card__icon {
    font-size: 25px;
}