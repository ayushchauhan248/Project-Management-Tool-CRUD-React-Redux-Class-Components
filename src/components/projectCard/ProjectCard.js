import "./ProjectCard.css";
import React, { Component } from "react";

class ProjectCard extends Component {
  render() {
    return (
      <div class="card">
        <p class="card__exit">
          <i class="fas fa-times">Title -</i>
        </p>
        <div class="card__icon">
          <i class="fas fa-bolt">{this.props.title}</i>
        </div>

        <p class="card__exit">
          <i class="fas fa-times">Technology -</i>
        </p>
        <div class="card__icon">
          <i class="fas fa-bolt">{this.props.technology}</i>
        </div>
      </div>
    );
  }
}

export default ProjectCard;
