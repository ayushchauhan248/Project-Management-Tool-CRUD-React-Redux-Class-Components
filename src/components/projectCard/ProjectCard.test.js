import React from "react";
import ProjectCard from "./ProjectCard";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

describe("Project tests", () => {
  it("snapshot testing", () => {
    const component = shallow(
      <ProjectCard title={"HELLO"} technology={"HIII"} />
    );
    expect(component).toMatchSnapshot();
  });

  it("Title text check", () => {
    const component = shallow(
      <ProjectCard title={"HELLO"} technology={"HIII"} />
    );
    expect(component.find("i").first().text()).toBe("Title -");
  });

  it("Technology text check", () => {
    const component = shallow(
      <ProjectCard title={"HELLO"} technology={"HIII"} />
    );
    expect(component.find("i").at(2).text()).toBe("Technology -");
  });
});
