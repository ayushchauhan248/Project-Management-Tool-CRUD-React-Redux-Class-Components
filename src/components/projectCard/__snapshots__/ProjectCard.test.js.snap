// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project tests snapshot testing 1`] = `
<div
  class="card"
>
  <p
    class="card__exit"
  >
    <i
      class="fas fa-times"
    >
      Title -
    </i>
  </p>
  <div
    class="card__icon"
  >
    <i
      class="fas fa-bolt"
    >
      HELLO
    </i>
  </div>
  <p
    class="card__exit"
  >
    <i
      class="fas fa-times"
    >
      Technology -
    </i>
  </p>
  <div
    class="card__icon"
  >
    <i
      class="fas fa-bolt"
    >
      HIII
    </i>
  </div>
</div>
`;
